### PR TITLE
MOM/JMS: Support for configurable m_subscriptionAwaitStartedSeconds

### PR DESCRIPTION
**Pull request by matthiaso**
323654

### CODE REVIEW

File: org.eclipse.scout.rt.mom.jms/src/main/java/org/eclipse/scout/rt/mom/jms/JmsMomImplementor.java [hunk 1]
```diff
@@ -128,6 +127,11 @@ public class JmsMomImplementor implements IMomImplementor {
    */
   public static final String JMS_REQUEST_CANCELLATION_MESSAGE_CONSUMER_JOB_RECEIVE_TIMEOUT = "scout.mom.jms.requestCancellationMessageConsumerJobReceiveTimeout";
 
+  /**
+   * Key to set {@link #m_subscriptionAwaitStartedSeconds}
+   */
```
Comment (paolobazzi) [inline org.eclipse.scout.rt.mom.jms/src/main/java/org/eclipse/scout/rt/mom/jms/JmsMomImplementor.java:134]: Maybe add a comment indicating that the absence of a configured value or a value of -1 or null means "infinite"

File: org.eclipse.scout.rt.mom.jms/src/main/java/org/eclipse/scout/rt/mom/jms/JmsMomImplementor.java [hunk 2]
```diff
@@ -291,7 +296,9 @@ protected JmsSessionProviderWrapper createSessionProviderWrapper(boolean transac
 
   protected JmsSubscription createJmsSubscription(IDestination<?> destination, SubscribeInput input, IJmsSessionProvider sessionProvider, IFuture<?> worker) {
     JmsSubscription subscription = new JmsSubscription(destination, input, sessionProvider, worker);
-    subscription.awaitStarted(m_subscriptionAwaitStartedSeconds, TimeUnit.SECONDS);
+    subscription.awaitStarted(
+        m_subscriptionAwaitStartedSeconds != -1 ? m_subscriptionAwaitStartedSeconds : null,
```
Comment (paolobazzi) [inline org.eclipse.scout.rt.mom.jms/src/main/java/org/eclipse/scout/rt/mom/jms/JmsMomImplementor.java]: Here -1 means 'infinite', in JmsSubscription instead a value of null indicated 'infinte'.
Suggestion: 
- Add constant public static int JmsMomImplementor.WAIT_TIME_INFINITE = -1;
- use constant as default value in JmsMomImplementor if no (or a null) config value is set in properties
- pass value as int to JmsSubscription
- change subscription to check for waitTime != JmsMomImplementor.WAIT_TIME_INFINITE to consider if the Thread should be blocked waiting for the subscription

File: org.eclipse.scout.rt.mom.jms/src/main/java/org/eclipse/scout/rt/mom/jms/JmsSubscription.java [hunk 3]
```diff
@@ -77,13 +77,13 @@ public ISubscriptionStats getStats() {
    *
    * @since 6.1
    */
-  public boolean awaitStarted(int time, TimeUnit unit) {
-    long timeoutNanos = System.nanoTime() + unit.toNanos(time);
+  public boolean awaitStarted(Integer time, TimeUnit unit) {
+    Long timeoutNanos = time != null && unit != null ? System.nanoTime() + unit.toNanos(time) : null;
```
Comment (paolobazzi) [inline org.eclipse.scout.rt.mom.jms/src/main/java/org/eclipse/scout/rt/mom/jms/JmsSubscription.java]: change signature back to int time & see comment in JmsMomImplementor

**Review by paolobazzi — COMMENTED** [review]
See comments for minor improvements

**Comment by matthiaso** [comment]
Pull request ready to be merged

**Pull request merged**